frontend/src/container/FeedItemWithPreview.jsx

```diff
@@ -324,8 +324,10 @@ export class FeedItemWithPreview extends React.Component {
         ? this.getFirstComment()
         : null
     )
+
+    const previewContent = props.content
     if (shouldShowComment && props.inRecentActivities) {
-      props.content.firstComment = commentToShow
+      previewContent.firstComment = commentToShow
```

Comment:
js variables are always assigned by reference. So this code doesn't fix the issue, you are still modifying props.content.firstComment

I'm not sure what you are trying to do but I expect something like this to solve your issue:
```jsx
const commentToDisplay = shouldShowComment && props.inRecentActivities
  ? commentToShow
  : props.content.firstComment
```

Reply:
but having variables "commentToDisplay" and "commentToShow" feels like it can be improved

Reply:
Due to the structure of `Comment`, I need to pass a copy of props.content to it as the `apiContent` prop. Changing how `Comment` works could be a solution but it could involve a handful of additional changes. I'll just edit the code to init `previewContent` as a deep copy of `props.content`

Reply:
Actually comment is only used twice, changing how comment works is way more performant than creating a deep copy